### PR TITLE
Add CNP/IP protocol support

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guidelines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

ISO/IEC 14908-4 defines how to tunnel Control Network Protocol (CNP) over IP networks. It encapsulates protocols like EIA-709, EIA-600, and CNP, making it a versatile solution for building automation and control systems.

### CODE REVIEW

**Comment by 0xA50C1A1** [comment]
@utoni Can you take a look, please?

**Comment by IvanNardi** [comment]
@0xA50C1A1, any chance do you have some traffic over TCP?

File: src/lib/protocols/cnp-ip.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  NDPI_LOG_DBG(ndpi_struct, "search CNP-IP\n");
+
+  if (packet->payload_packet_len > 20 && packet->payload_packet_len == ntohs(get_u_int16_t(packet->payload, 0)) &&
+      (packet->payload[3] & 0x1F) == 0x01 && (packet->payload[5] & 0x1F) < 0x03)
+  {
+    u_int32_t session_id = ntohl(get_u_int32_t(packet->payload, 8));
+
+    if (!flow->cnp_ip_stage) {
+      flow->cnp_ip_stage = 1;
+      flow->cnp_ip_session_id = session_id;
```
Comment (utoni) [inline src/lib/protocols/cnp-ip.c]: Is it possible to skip either `cnp_ip_stage` or `cnp_ip_session_id` for a successful classification?

Comment (utoni) [reply src/lib/protocols/cnp-ip.c]: (to reduce the size of flow struct)

**Comment by 0xA50C1A1** [comment]
> @0xA50C1A1, any chance do you have some traffic over TCP?

Unfortunately, no.

File: src/lib/protocols/cnp-ip.c (hunk 1 above)
Comment (0xA50C1A1) [reply src/lib/protocols/cnp-ip.c]: > Is it possible to skip either `cnp_ip_stage` or `cnp_ip_session_id` for a successful classification?




I guess I could replace this by checking the `Packet Type` field, but that would reduce the detection accuracy.

**Review by utoni — APPROVED** [review]
Great work, LGTM!

**Comment by 0xA50C1A1** [comment]
@IvanNardi @utoni I've simplified the check, it seems to be working. If false positives occur, I can revert the `SessionID` check.

**Review by IvanNardi — APPROVED** [review]

**Pull request merged**